bindings/node.js/kzg.cxx

```diff
@@ -0,0 +1,341 @@
+    return env.Null();
+  }
+
+  bool verificationResult;
+  if (verify_aggregate_kzg_proof(
+    &verificationResult,
+    polynomial,
+    commitments,
+    numberOfBlobs,
+    &proof,
+    kzgSettings
+  ) != C_KZG_OK) {
+    free(commitments);
+    free(polynomial);
+
+    Napi::TypeError::New(env, "Error calling verify_aggregate_kzg_proof")
+      .ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  free(commitments);
+  free(polynomial);
+
+  return Napi::Boolean::New(env, verificationResult);
+}
+
+// verifyKzgProof: (polynomialKzg: KZGCommitment, z: BLSFieldElement, y: BLSFieldElement, kzgProof: KZGProof, setupHandle: SetupHandle) => boolean;
+Napi::Value VerifyKzgProof(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  if (info.Length() != 5) {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+      .ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  auto c_param = info[0].As<Napi::TypedArray>();
+  if (c_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(env, "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+  auto polynomialKzg = c_param.As<Napi::Uint8Array>().Data();
+
+  auto z_param = info[0].As<Napi::TypedArray>();
+  if (z_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(env, "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+  auto z = z_param.As<Napi::Uint8Array>().Data();
+
+  auto y_param = info[0].As<Napi::TypedArray>();
+  if (y_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(env, "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+  auto y = y_param.As<Napi::Uint8Array>().Data();
+
+  auto proof_param = info[0].As<Napi::TypedArray>();
+  if (proof_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(info.Env(), "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return info.Env().Undefined();
+  }
+  auto kzgProof = proof_param.As<Napi::Uint8Array>().Data();
+
+  auto kzgSettings = info[4].As<Napi::External<KZGSettings>>().Data();
+
+  BLSFieldElement fz, fy;
+  bytes_to_bls_field(&fz, z);
+  bytes_to_bls_field(&fy, y);
+
+  KZGCommitment commitment;
+  auto ret = bytes_to_g1(&commitment, polynomialKzg);
+  if (ret != C_KZG_OK) {
+    std::ostringstream ss;
+    std::copy(polynomialKzg, polynomialKzg + BYTES_PER_COMMITMENT, std::ostream_iterator<int>(ss, ","));
+
+    Napi::TypeError::New(env, "Failed to parse argument commitment: "  + ss.str() + " Return code was: " + std::to_string(ret)).ThrowAsJavaScriptException();
+    return env.Null();
+  };
+
+  KZGProof proof;
+  if (bytes_to_g1(&proof, kzgProof) != C_KZG_OK) {
+    Napi::TypeError::New(env, "Invalid kzgProof").ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  bool out;
+  if (verify_kzg_proof(&out, &commitment, &fz, &fy, &proof, kzgSettings) == C_KZG_OK) {
```

Comment:
This seems like it's returning `C_KZG_OK` when it shouldn't.

Reply:
@xrchz My TypeScript test driving this passes 32 0-bytes for z and y and it passes. It seems like it passes with any values of z and y... I must have misunderstood something.

Reply:
I'm not using the `out` result correctly. Will fix.